components/brave_wallet/browser/solana_message.cc

```diff
@@ -719,4 +704,113 @@ bool SolanaMessage::UsesDurableNonce() const {
   return true;
 }
 
+bool SolanaMessage::UsesPriorityFee() const {
+  if (instructions_.empty()) {
+    return false;
+  }
+
+  auto instruction = instructions_[0];
+
+  // If the first instruction is advance nonce instruction,
+  // use the second instruction.
+  if (solana_ins_data_decoder::GetSystemInstructionType(
+          instruction.data(), instruction.GetProgramId()) ==
+      mojom::SolanaSystemInstruction::kAdvanceNonceAccount) {
+    if (instructions_.size() < 2) {
+      return false;
+    }
+
+    instruction = instructions_[1];
+  }
+
+  auto instruction_type =
+      solana_ins_data_decoder::GetComputeBudgetInstructionType(
+          instruction.data(), instruction.GetProgramId());
+  if (!instruction_type) {
+    return false;
+  }
+
+  return instruction_type ==
+         mojom::SolanaComputeBudgetInstruction::kSetComputeUnitLimit;
+}
```

Comment:
From https://www.helius.dev/blog/priority-fees-understanding-solanas-transaction-fee-mechanics#how-to-implement-priority-fees-programmatically.
>A transaction’s prioritization fee is set by setting a SetComputeUnitPrice instruction and an optional SetComputeUnitLimit instruction. If a SetComputeUnitPrice instruction isn’t provided, the transaction will default to the lowest priority since no additional fee is provided.

Maybe we should look over all instructions and see if there's SetComputeUnitPrice instead? Is there a rule that this instruction needs to be in the first instruction (excluding advance nonce ins)?

Reply:
I couldn't find documentation that definitely says it's required, but it's at least best practice.  Seems possible transactions would fail otherwise. Either way, looking over all instructions seems fine to me too.